### PR TITLE
[Dynamic dashboard] New widget card style

### PR DESCRIPTION
**Pull request by 0nko**
Part of #11314. This PR implements a new dynamic dashboard widget card style that contains a common header with a menu and a footer (optional button). The actual code doesn't change anything visible yet (more to come in the next PRs) to limit the amount of changes.

image

**To test:**
1. Start the app with the dynamic dashboard feature flag disabled
2. Smoke test it and verify things are working as expected (Blaze card and stats card)

**Note: ** The PR can be merged once the target branch is `trunk`

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/DashboardContainer.kt [hunk 1]
```diff
@@ -57,56 +61,45 @@ private fun WidgetList(
     Column(
         modifier = Modifier
             .fillMaxSize()
+            .background(MaterialTheme.colors.surface)
     ) {
         widgets.forEach {
             AnimatedVisibility(it.isVisible) {
                 Column {
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/DashboardContainer.kt:68]: Np. Is the purpose of this `Column` to just add a `Spacer` below each widget? If that is the case we can use `verticalArrangement = Arrangement.spacedBy(16.dp)` in the parent column in line 61, and get rid of this one and the Spacer below.

**Review by JorgeMucientes — APPROVED** [review]

**Comment by JorgeMucientes** [comment]
Great job @0nko, the old code works as expected and the new container feature will be handy to migrate all the widgets on the screen

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/OverflowMenu.kt [hunk 2]
```diff
@@ -27,18 +27,18 @@ import com.woocommerce.android.R.string
 @Composable
 fun <T> WCOverflowMenu(
     items: Array<T>,
-    onSelected: (T) -> Unit,
-    mapper: (T) -> String = { it.toString() },
     modifier: Modifier = Modifier,
+    onSelected: (T) -> Unit,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/component/OverflowMenu.kt:31]: np, before merging this, we might want to put the optional argument below the non-optional one.

File: WooCommerce/src/main/res/layout/fragment_my_store.xml [hunk 3]
```diff
@@ -58,7 +58,6 @@
                     <!-- Order stats -->
                     <com.woocommerce.android.ui.mystore.MyStoreStatsView
                         android:id="@+id/my_store_stats"
-                        style="@style/Woo.Card"
```
Comment (hichamboushaba) [inline WooCommerce/src/main/res/layout/fragment_my_store.xml:61]: Is this safe and needed? this is done on the legacy screen.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WidgetCard.kt [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
+import com.woocommerce.android.ui.dashboard.DashboardViewModel.DashboardWidgetAction
+import com.woocommerce.android.ui.dashboard.DashboardViewModel.DashboardWidgetMenu
+
+@Composable
+fun WidgetCard(
+    modifier: Modifier = Modifier,
+    @StringRes titleResource: Int,
+    @DrawableRes iconResource: Int? = null,
+    menu: DashboardWidgetMenu,
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WidgetCard.kt:36]: It seems the only option for the overflow we'll have is `hide {title}` for all p1712742336085579/1712313037.725699-slack-C03L1NF1EA3, we can reduce duplication by setting the menu directly here in the card, WDYT? Unless you think that we'll need to customize the menus in the future, but generally I would vote to not add it (**YAGNI**)

**Comment by hichamboushaba** [comment]
@JorgeMucientes I know the PR is already approved, but as I need to build on top of this to update the style of the stats card, I left some comments above, and I pushed a commit cd6c4f5 to fix the order of some arguments to avoid having merge conflicts later.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WidgetCard.kt (hunk 4 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WidgetCard.kt:36]: Hmm, @hichamboushaba I'm not sure about that one. The onboarding widget currently has 2 actions: `hide` and `feedback`.

Screenshot 2024-04-18 at 18 12 36

Imho I'd keep the flexibility for each widget to ad its own menu items. I think neither @itsmeichigo or Joe where aware of these 2 actions when discussing that in the Slack convo you linked.

Comment (hichamboushaba) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/dashboard/WidgetCard.kt:36]: Nice find @JorgeMucientes, I didn't see this part, and I agree with you.

File: WooCommerce/src/main/res/layout/fragment_my_store.xml (hunk 3 above)
Comment (0nko) [reply WooCommerce/src/main/res/layout/fragment_my_store.xml:61]: This was removed when I started working on moving the Stat card to the new model. I decided to split the work into multiple PRs but forgot about this.

**Pull request merged**